cmd/juju/action/cancel.go

```diff
@@ -88,35 +88,36 @@ func (c *cancelCommand) Run(ctx *cmd.Context) error {
 			return errors.Errorf("no actions found matching prefix %s, no actions have been canceled", requestedID)
 		}
 
-		actionTags = append(actionTags, requestedActionTags...)
-	}
-
-	entities := []params.Entity{}
-	for _, tag := range actionTags {
-		entities = append(entities, params.Entity{Tag: tag.String()})
+		for _, tag := range requestedActionTags {
+			actionIDs = append(actionIDs, tag.Id())
+		}
+		idsToCancel = actionIDs
 	}
 
-	actions, err := api.Cancel(params.Entities{Entities: entities})
+	actions, err := api.Cancel(idsToCancel)
 	if err != nil {
 		return err
 	}
 
-	if len(actions.Results) < 1 {
-		return errors.Errorf("identifier(s) %q matched action(s) %q, but no actions were canceled", c.requestedIDs, actionTags)
+	if len(actions) < 1 {
+		if len(actionIDs) > 0 {
+			return errors.Errorf("identifier(s) %q matched action(s) %q, but no actions were canceled", c.requestedIDs, actionIDs)
+		}
+		return errors.New("no actions were canceled")
 	}
 
 	type unCanceledAction struct {
-		ActionTag names.ActionTag
-		Result    *params.ActionResult
+		ID     string
+		Result *actionapi.ActionResult
 	}
 	var unCanceledActions []unCanceledAction
```

Comment:
Can we please rename this variable. unCanceledActions isn't quite right and just looks wrong.

Perhaps failedCancels.

Reply:
ok